breadth_first_search/BreadthFirstSearch.java

```diff
@@ -0,0 +1,81 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+//Generic type BreadthFirstSearch implementation
+public class BreadthFirstSearch<T> {
+    // HasMap  of lists for Adjacency List Representation
+    public HashMap<T, ArrayList<T>> adj = new HashMap<T, ArrayList<T>> ();
+   
```

Comment:
@jsroyal You have to remove whitespace after line 10 and 25 here. 

```
the line has trailing whitespace             
this line does not
```

⬆️  Select the contents of block above to understand.